tortoise/backends/base/schema_generator.py

```diff
@@ -179,23 +179,26 @@ def _get_table_sql(self, model: "Type[Model]", safe: bool = True) -> dict:
                     if reference.description
                     else ""
                 )
+
+                field_type = reference.to_field.get_for_dialect(self.DIALECT, "SQL_TYPE")
```

Comment:
Good catch, I didn't think about this :+1:
The only thing I'm concerned about is that `to_field` mutates? It starts of as an `Optional[str]` and then mutates to `Field`.

Reply:
Thank you for review. I think so and also CI catch that point(logs are below). So how can we fix this? I think pass `Field` to schema_generator is inevitable. Can we type case to_field or do we have another workaround?

``` bat
tortoise/backends/base/schema_generator.py:183: error: Item "str" of
"Optional[str]" has no attribute "get_for_dialect"
                    field_type = reference.to_field.get_for_dialect(self.D...
                                 ^
tortoise/backends/base/schema_generator.py:183: error: Item "None" of
"Optional[str]" has no attribute "get_for_dialect"
                    field_type = reference.to_field.get_for_dialect(self.D...
...
```

Reply:
I think lets not mutate the type.

Leave the ``Optional[str]`` ``to_field`` in the Field, but only use it as read only.
Add a new field that is for internal use, like https://github.com/tortoise/tortoise-orm/blob/develop/tortoise/fields/relational.py#L280-L285
possibly called ``to_field_instance``?

Consider moving the ``to_field`` to the ``RelationalField`` base class (which is a new common base class to fix some typing in a recent PR) to reduce duplication of code.

Reply:
Oh, I see. That's really good opinion. I will make patch for that. I also check your above checkpoint with PostgreSQL UUIDField and i work well.

Reply:
:+1: